# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: 88a9fe04aaa5
# Full Hash: 88a9fe04aaa582676a84734d67c9069b68a7f3e1
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Update Audio- and VideoConduit stats API and integration. r=ng,bwc
#   
#   For AudioConduit this patch:
#   - Uses correct threads per upstream, pending changes due to the threading model
#     update that's coming
# ==============================================================================

diff -r 3eae9f9279f6 -r 88a9fe04aaa5 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Fri Jan 29 15:38:54 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Mon Jan 18 15:28:29 2021 +0100
@@ -179,21 +179,9 @@
 
   void SetSyncGroup(const std::string& group) override;
 
-  bool GetSendPacketTypeStats(
-      webrtc::RtcpPacketTypeCounter* aPacketCounts) override;
-
-  bool GetRecvPacketTypeStats(
-      webrtc::RtcpPacketTypeCounter* aPacketCounts) override;
-
-  bool GetRTPReceiverStats(unsigned int* jitterMs,
-                           unsigned int* cumulativeLost) override;
-  bool GetRTCPReceiverReport(uint32_t* jitterMs, uint32_t* packetsReceived,
-                             uint64_t* bytesReceived, uint32_t* cumulativeLost,
-                             Maybe<double>* aOutRttSec) override;
-  bool GetRTCPSenderReport(unsigned int* packetsSent, uint64_t* bytesSent,
-                           DOMHighResTimeStamp* aRemoteTimestamp) override;
-  Maybe<mozilla::dom::RTCBandwidthEstimationInternal> GetBandwidthEstimation()
-      override;
+  Maybe<webrtc::AudioReceiveStream::Stats> GetReceiverStats() const override;
+  Maybe<webrtc::AudioSendStream::Stats> GetSenderStats() const override;
+  webrtc::Call::Stats GetCallStats() const override;
 
   bool SetDtmfPayloadType(unsigned char type, int freq) override;
 